### PR TITLE
add `cupy.random.wald`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: cupy/random/distributions.py [hunk 1]
```diff
@@ -418,3 +418,32 @@ def vonmises(mu, kappa, size=None, dtype=float):
...
+
+def wald(mean, scale, size=None, dtype=float):
+    """Wald distribution.
+
+    Returns an array of samples drawn from the wald distribution. Its
+    probability density function is defined as
+
+    .. math::
+       f(x) = \\sqrt{\\frac{scale}{2\\pi x^3}}\\
+           e^{\\frac{-scale(x-mean)^2}{2mean^2x}},
```
Comment (toslunar) [inline cupy/random/distributions.py]: Why don't you use `\lambda` and `\mu` to denote scale and mean?

File: cupy/random/distributions.py [hunk 2]
```diff
@@ -418,3 +418,32 @@ def vonmises(mu, kappa, size=None, dtype=float):
     """
     rs = generator.get_random_state()
     return rs.vonmises(mu, kappa, size=size, dtype=dtype)
+
+
+def wald(mean, scale, size=None, dtype=float):
+    """Wald distribution.
+
+    Returns an array of samples drawn from the wald distribution. Its
```
Comment (toslunar) [inline cupy/random/distributions.py]: Could you change `wald` to `Wald` in the documents?

File: cupy/random/generator.py [hunk 3]
```diff
@@ -468,6 +468,34 @@ def vonmises(self, mu, kappa, size=None, dtype=float):
...
+        'wald_scale')
+
+    def wald(self, mean, scale, size=None, dtype=float):
+        """Returns an array of samples drawn from the wald distribution.
+
+         .. seealso::
+            :func:`cupy.random.wald` for full documentation,
+            :meth:`numpy.random.RandomState.wald`
+        """
+        mean, scale = cupy.asarray(mean), cupy.asarray(scale)
```
Comment (toslunar) [inline cupy/random/generator.py]: Could you fix `size` if it's `None`?  (#1644)

File: cupy/random/distributions.py (hunk 1 above)
Comment (YoshikawaMasashi) [reply cupy/random/distributions.py]: I imitated numpy document. but I'll fix it.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: cupy/random/generator.py [hunk 4]
```diff
@@ -536,6 +536,36 @@ def vonmises(self, mu, kappa, size=None, dtype=float):
...
+            X = mean + mu_2l*(Z - sqrt(4*scale*Z + Z*Z));
+            if (! U <= mean/(mean+X))
+            {
+                X = mean*mean/X;
+            }
+        """,
+        'wald_scale')
+
+    def wald(self, mean, scale, size=None, dtype=float):
+        """Returns an array of samples drawn from the wald distribution.
```
Comment (toslunar) [inline cupy/random/generator.py]: Could you fix `wald` to `Wald` in this line, too?

File: cupy/random/generator.py [hunk 5]
```diff
@@ -536,6 +536,36 @@ def vonmises(self, mu, kappa, size=None, dtype=float):
...
+         .. seealso::
+            :func:`cupy.random.wald` for full documentation,
+            :meth:`numpy.random.RandomState.wald`
+        """
+        mean, scale = cupy.asarray(mean), cupy.asarray(scale)
+        if size is None:
+            size = cupy.broadcast(mean, scale).shape
+        x = self.normal(size=size, dtype=dtype)
+        u = self.random_sample(size=size, dtype=dtype)
+        RandomState._wald_kernel(mean, scale, u, x, x)
```
Comment (toslunar) [inline cupy/random/generator.py]: It seems returning the output of the kernel is ok.

**Comment by YoshikawaMasashi** [comment]
sorry, I found this error.
```
>>> np.random.wald(1, 1, size=10)
array([0.57100968, 1.17194574, 0.35100232, 0.71970238, 0.26104351,
       0.80307992, 1.04960889, 1.88728334, 0.90305322, 1.48224124])
>>> cp.random.wald(1, 1, size=10)
array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```

**Comment by toslunar** [comment]
It seems `490e0a8` fixed it.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**